src/main/java/io/confluent/idesidecar/restapi/util/SanitizeHeadersUtil.java

```diff
@@ -0,0 +1,36 @@
+package io.confluent.idesidecar.restapi.util;
+
+import io.vertx.core.MultiMap;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * Utility class for sanitizing HTTP headers.
+ */
+public final class SanitizeHeadersUtil {
+
+  private SanitizeHeadersUtil() {
+    // Private constructor to prevent instantiation
+  }
+
+  /**
+   * Sanitizes HTTP headers by removing excluded headers.
+   *
+   * @param requestHeaders The headers to sanitize
+   * @param exclusions List of header names to exclude
+   * @return Map of sanitized headers
+   */
+  public static Map<String, String> sanitizeHeaders(MultiMap requestHeaders, List<String> exclusions) {
+    var headers = new HashMap<String, String>();
+    if (requestHeaders != null) {
```

Comment:
If requestHeaders is null, calling this method should throw an `IllegalArgumentException` or similar.